app/updater.js

```diff
@@ -84,21 +85,26 @@ var scheduleUpdates = () => {
 }
 
 // set the feed url for the auto-update system
-exports.init = (platform, arch, ver) => {
+exports.init = (platform, arch, ver, updateToPreview) => {
   // When starting up we should not expect an update to be available
   appActions.setUpdateStatus(UpdateStatus.UPDATE_NONE)
 
   // Browser version X.X.X
   version = ver
 
+  // Flag controlling whether preview releases are accepted
+  updateToPreviewReleases = updateToPreview
+
   var baseUrl = exports.updateUrl(appConfig.updates, platform, arch)
+  var query = { accept_preview: updateToPreviewReleases ? 'true' : 'false' }
 
   if (baseUrl) {
     debug(`updateUrl = ${baseUrl}`)
     scheduleUpdates()
     // This will fail if we are in dev
     try {
-      autoUpdater.setFeedURL(baseUrl)
+      // add the preview flag to the base feed url
+      autoUpdater.setFeedURL(`${baseUrl}?${querystring.stringify(query)}`)
```

Comment:
or even cleaner just ?accept_preview=true or ?accept_preview=false, but that might require server change.